toughday/src/main/java/com/adobe/qe/toughday/internal/core/engine/runmodes/ConstantLoad.java

```diff
@@ -52,22 +57,32 @@
     private long interval = DEFAULT_INTERVAL;
     private int rate;
     private int currentLoad;
+    //private int currentLoad;
+    private long initialDelay = 0;
+    /* field used for checking the finish condition when running TD distributed with the rate
+     * smaller than number of agents in the cluster. */
+    private int oneAgentRate = 0;
 
+    private ScheduledExecutorService runRoundScheduler = Executors.newScheduledThreadPool(1);
     private TestCache testCache;
     private Phase phase;
+    private RunModeSplitter<ConstantLoad> runModeSplitter = new ConstantLoadRunModeSplitter();
+    private ScheduledFuture<?> scheduledFuture = null;
+    private final ConstantLoadRunModeBalancer runModeBalancer = new ConstantLoadRunModeBalancer();
 
     private Boolean measurable = true;
 
-    @ConfigArgSet(required = false, defaultValue = DEFAULT_LOAD_STRING, desc = "Set the load, in requests per second for the \"constantload\" runmode.")
+    @ConfigArgSet(required = false, defaultValue = DEFAULT_LOAD_STRING,
+            desc = "Set the load, in requests per second for the \"constantload\" runmodes.")
     public void setLoad(String load) {
         checkNotNegative(Long.parseLong(load), "load");
         this.load = Integer.parseInt(load);
     }
 
-    @ConfigArgGet
+    @ConfigArgGet(redistribute = true)
```

Comment:
The run mode balancer knows how to interpret the redistribution instructions received from the Driver and how to modify the properties of the run mode instance so that the desired configuration will be achieved (it adapts the number of threads or the current load, etc).

We need the  `redistribute` field to automatically detect which properties should be included in the redistribution instruction message when creating it.